### PR TITLE
fix: LP error message improvements when zero liquidity

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

Smarter error message when LP ranges get out of range near bounds due to significant price movements. Prompted by the TIA launch

### CODE REVIEW

File: x/concentrated-liquidity/lp.go [hunk 1]
```diff
@@ -107,7 +108,21 @@ func (k Keeper) CreatePosition(ctx sdk.Context, poolId uint64, owner sdk.AccAddr
...
+			If range becomes inactive before getting on chain, you might need to provide more of one token as opposed to two of the original amount`, amount0Desired, amount1Desired)
+		} else if amount0Desired.IsZero() {
+			return CreatePositionData{}, fmt.Errorf(`failed to translate amount1 (%d) to positive liquidity. Either try providing more liquidity or confirm that given range is still inactive.
+			If the given range becomes activated, two tokens will be needed as opposed to one.`, amount1Desired)
+		}
+
+		// amount1Desired is zero
+
+		return CreatePositionData{}, fmt.Errorf(`failed to translate amount0 (%d) to positive liquidity. Either try providing more liquidity or confirm that given range is still inactive.
+			If the given range becomes activated, two tokens will be needed as opposed to one.`, amount0Desired)
```
Comment (czarcas7ic) [inline x/concentrated-liquidity/lp.go:125]: Do you think we should just explicitly check that amount1Desired.IsZero() here and then have an else branch that is "unknown error" for why liquidityDelta is zero? Just might come to bite us later

**Review by czarcas7ic — APPROVED** [review]
LGTM, just small opinionated comment

File: x/concentrated-liquidity/lp.go (hunk 1 above)
Comment (p0mvn) [reply x/concentrated-liquidity/lp.go:125]: If we construct a boolean table, we do check all logic branches though so just `else` is fine IMO

See this:
https://github.com/osmosis-labs/osmosis/blob/41100c97c5b0a1259e0404b637bc1b3656cc9103/x/concentrated-liquidity/lp.go#L68-L70
- true & true

https://github.com/osmosis-labs/osmosis/blob/41100c97c5b0a1259e0404b637bc1b3656cc9103/x/concentrated-liquidity/lp.go#L113-L115
- false & false

https://github.com/osmosis-labs/osmosis/blob/41100c97c5b0a1259e0404b637bc1b3656cc9103/x/concentrated-liquidity/lp.go#L116-L119
true and false

so `else` must be false and true:
https://github.com/osmosis-labs/osmosis/blob/41100c97c5b0a1259e0404b637bc1b3656cc9103/x/concentrated-liquidity/lp.go#L123-L124

Comment (AlpinYukseloglu) [reply x/concentrated-liquidity/lp.go:124]: Either this, or maybe slightly relaxing the language to something like "Possible reasons could be..."

**Review by AlpinYukseloglu — APPROVED** [review]
LGTM, agree with adam's comment (see response) but not blocking

**Comment by p0mvn** [comment]
Updated per request by relaxing the language @czarcas7ic @AlpinYukseloglu 

[5e1a2fd](https://github.com/osmosis-labs/osmosis/pull/6794/commits/5e1a2fdf6b6efba3c7f987c4990aa67aaf2f21d0)

**Pull request merged**